### ISSUE TITLE
Screen Reader: Global: Selected option in option list is not announced as selected

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77389]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS or TalkBack on Android, open the app
2. Swipe right to focus the Account tab and double-tap to activate it
3. Swipe right to focus the Profile button and double-tap to activate it
4. Swipe right to focus the Status menu item and double-tap to activate it
5. Swipe right to focus the Clear after menu item and double-tap to activate it
6. Swipe right to focus the currently selected item and note how it is announced

This is a global issue affecting all selected items in item lists
Examples:
1. Status: Vacation Delegate
2. Preferences: Language
3. Preferences: Priority Mode
4. Profile: Country
5. Profile: Timezone
6. Wallet: Add Bank Account
7. Create Expense: Category
8. Create Workspace > Company Cards: Add Card
9. Profile > Status > Emoji Status > Change default skin tone
10. New chat > Select emoji > Change default skin tone
11. Group chat > Select emoji > Change default skin tone

## Expected Result:
When navigating a list of options, the currently selected option should be announced as selected, e.g.:
"1 hour, selected, button"
"Never, selected, button"

The user should always know which option is currently active without relying on visuals.

## Actual Result:
The selected item is only visually highlighted, while screen readers do not announce it as selected.
The list item is announced only by its name, with no selected state information:
"After 1 hour, button"
"Never, button"

Screen reader users cannot tell which option is currently active.

## Area issue was found in:
Profile > Status

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Add the accessibilityState prop to the element to ensure it is properly announced, for example:
"1 hour, selected, button"
"Never, selected, button"
https://appt.org/en/docs/react-native/samples/accessibility-state
https://reactnative.dev/docs/accessibility#accessibilitystate

## Environment(s)
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos
<img width="1179" height="2424" alt="Image" src="https://github.com/user-attachments/assets/2a83c2b4-ee4d-444b-8cc4-1f4aa0b0ce57" />

https://github.com/user-attachments/assets/dce4261d-0bae-4531-9b4d-0085dba48f78

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7006885
**Issue reported by:** NA
**Slack conversation:** NA
